chainntnfs/btcdnotify/btcd.go

```diff
@@ -403,83 +389,59 @@ out:
 	b.wg.Done()
 }
 
-// attemptHistoricalDispatch tries to use historical information to decide if a
-// notification ca be dispatched immediately, or is partially confirmed so it
-// can skip straight to the confirmations heap.
-//
-// Returns true if the transaction was either partially or completely confirmed
-func (b *BtcdNotifier) attemptHistoricalDispatch(
-	msg *confirmationsNotification) bool {
-
-	chainntnfs.Log.Infof("Attempting to trigger dispatch for %v from "+
-		"historical chain", msg.txid)
+// historicalConfDetails looks up whether a transaction is already included in a
+// block in the active chain and, if so, returns details about the confirmation.
+func (b *BtcdNotifier) historicalConfDetails(txid *chainhash.Hash,
+) (*chainntnfs.TxConfirmation, error) {
 
 	// If the transaction already has some or all of the confirmations,
 	// then we may be able to dispatch it immediately.
-	tx, err := b.chainConn.GetRawTransactionVerbose(msg.txid)
+	tx, err := b.chainConn.GetRawTransactionVerbose(txid)
 	if err != nil || tx == nil || tx.BlockHash == "" {
-		jsonErr, ok := err.(*btcjson.RPCError)
-		switch {
-		case ok && jsonErr.Code == -5:
-		default:
-			chainntnfs.Log.Warnf("unable to query for txid(%v): %v",
-				msg.txid, err)
+		if err == nil {
+			return nil, nil
+		}
+		if jsonErr, ok := err.(*btcjson.RPCError); ok && jsonErr.Code == -5 {
```

Comment:
Can you add a comment here to indicate that this means the transaction wasn't yet found in the mempool? Thanks!